apps/nextjs/src/stores/lessonPlanStore/stateActionFunctions/handleTrackingEvents.ts

```diff
@@ -0,0 +1,25 @@
+import invariant from "tiny-invariant";
+
+import type { LessonPlanTrackingContextProps } from "@/lib/analytics/lessonPlanTrackingContext";
+
+import type { LessonPlanStore } from "..";
+
+export const handleTrackingEvents = (
+  lessonPlanTracking: LessonPlanTrackingContextProps,
+  get: () => LessonPlanStore,
+) => {
+  const { chatActions } = get();
+
+  const messages = chatActions?.getMessages();
```

Comment:
I think this is a hint that we should tweak our pattern here. Maybe we need something like `chatStore.getState().stableMessages` or `getChatStore().stableMessages` so that we can access the latest version of the store directly?